python/rascaline/rascaline/utils/clebsch_gordan/_correlate_density.py

```diff
@@ -178,11 +192,12 @@ def __init__(
         )
         # Parse the bool flags that control skipping of redundant CG combinations
         # and TensorMap output from each iteration
-        self._skip_redundant, self._output_selection = (
+        self._skip_redundant, self._output_selection, self._keep_l_in_keys = (
             _utils.parse_bool_iteration_filters(
                 n_iterations,
                 skip_redundant=skip_redundant,
                 output_selection=output_selection,
+                keep_l_in_keys=keep_l_in_keys,
             )
```

Comment:
why are we parsing multiple thing in a single function call? Instead of 
```suggestion
        self._skip_redundant = _utils.parse_bool_iteration_filters(n_iterations, skip_redundant)
        self._output_selection = _utils.parse_bool_iteration_filters(n_iterations, output_selection)
        self._keep_l_in_keys = _utils.parse_bool_iteration_filters(n_iterations, keep_l_in_keys)
```

Reply:
Main reason is because they (and their defaults) interact in some way

Reply:
It seems that they don't interact actually. `parse_bool_iteration_filters` is just handling the three of them one after the other.

Reply:
Sure, done